### ISSUE TITLE
k8s_yaml handles string yaml poorly

### ISSUE THREAD

**Issue by landism** [windmilleng/tilt#2386]
repro:
make a 1-line tiltfile with `k8s_yaml(str(local('kustomize build https://github.com/kubernetes-sigs/cluster-api//config/default?ref=v0.2.5')))`
(Note this is a bad Tiltfile - the `str` should not be there. I wrote this while mistakenly thinking I needed to convert my blob to a string to pass it to `k8s_yaml`)

observed:
fsevent errors that screw up the terminal
![image](https://user-images.githubusercontent.com/7453991/67221789-10c17c00-f3fa-11e9-9a13-296a77d6deb9.png)

expected:
not that

This is handled [here](https://github.com/windmilleng/tilt/blob/2fa8c1142a547325e25a6141ce3755fad1f42bc0/internal/tiltfile/k8s.go#L690).

Basically, absPathFromStarlarkValue happily converts this huge yaml string into an absolute path, and then we try to open it and set up watches on it, and then apparently fsevents generate terminal-destroying log messages when it get bad paths?

So maybe there are a couple things here:
1. fsevents errors shouldn't garble the terminal
2. We've talked about inferring string vs blob - it'd be nice to give a better error message here. (even ignoring #1, the log ends up being so much yaml that you can't see the start of the log, and then a "filename too long", and it's not super clear what happened)
